netmaster/mastercfg/networkstate.go

```diff
@@ -43,19 +43,24 @@ const (
 // vlans with ovs. The state is stored as Json objects.
 type CfgNetworkState struct {
 	core.CommonState
-	Tenant      string        `json:"tenant"`
-	NetworkName string        `json:"networkName"`
-	NwType      string        `json:"nwType"`
-	PktTagType  string        `json:"pktTagType"`
-	PktTag      int           `json:"pktTag"`
-	ExtPktTag   int           `json:"extPktTag"`
-	SubnetIP    string        `json:"subnetIP"`
-	SubnetLen   uint          `json:"subnetLen"`
-	Gateway     string        `json:"gateway"`
-	EpAddrCount int           `json:"epAddrCount"`
-	EpCount     int           `json:"epCount"`
-	IPAllocMap  bitset.BitSet `json:"ipAllocMap"`
-	DNSServer   string        `json:"dnsServer"`
+	Tenant        string          `json:"tenant"`
+	NetworkName   string          `json:"networkName"`
+	NwType        string          `json:"nwType"`
+	PktTagType    string          `json:"pktTagType"`
+	PktTag        int             `json:"pktTag"`
+	ExtPktTag     int             `json:"extPktTag"`
+	SubnetIP      string          `json:"subnetIP"`
+	SubnetLen     uint            `json:"subnetLen"`
+	Gateway       string          `json:"gateway"`
+	EpAddrCount   int             `json:"epAddrCount"`
+	EpCount       int             `json:"epCount"`
+	IPAllocMap    bitset.BitSet   `json:"ipAllocMap"`
+	DNSServer     string          `json:"dnsServer"`
+	IPv6Subnet    string          `json:"ipv6SubnetIP"`
+	IPv6SubnetLen uint            `json:"ipv6SubnetLen"`
+	IPv6Gateway   string          `json:"ipv6Gateway"`
+	IPv6AllocMap  map[string]bool `json:"ipv6AllocMap"`
```

Comment:
Just a discussion point:
Any idea about performance implication of keeping all IP addresses in one etcd object?
Also, Whats the allocation performance since we have to do sequential walk to find next available address